godot-core/src/builtin/dictionary.rs

```diff
@@ -0,0 +1,314 @@
+use godot_ffi as sys;
+
+use crate::builtin::{inner, FromVariant, ToVariant, Variant, VariantConversionError};
+use std::collections::{HashMap, HashSet};
+use std::fmt;
+use sys::{ffi_methods, interface_fn, GodotFfi};
+use sys::types::*;
+
+use super::Array;
+
+/// Godot's `Dictionary` type.
+///
+/// The keys and values of the array are all `Variant`, so they can all be of different types.
+/// 
+/// Keys are assumed to be cheap to clone, this will usually be the case especially for 
+/// value types and reference counted types.
+///
+/// # Safety
+///
+/// TODO: I suspect it is similar to the rules for array.
```

Comment:
As I understood it, yes. Maybe instead of copy-pasting that section, link to `Array`:
```rs
/// # Thread safety
///
/// The same principles apply as for [`Array`]. Consult its documentation for details.
```